gnome-initial-setup/eos/hack-sound-client/eos-hack-sound-client.c

```diff
@@ -0,0 +1,211 @@
+}
+
+static void
+eos_hack_sound_client_init (EOSHackSoundClient *client)
+{
+  /* empty */
+}
+
+static void
+eos_hack_sound_client_finalize (GObject *object)
+{
+  EOSHackSoundClient *client = EOS_HACK_SOUND_CLIENT (object);
+  EOSHackSoundClientPrivate *priv =
+      eos_hack_sound_client_get_instance_private (client);
+
+  if (priv->proxy != NULL)
+    g_object_unref (priv->proxy);
+  G_OBJECT_CLASS (eos_hack_sound_client_parent_class)->finalize (object);
+}
+
+void
+eos_hack_sound_client_stop (EOSHackSoundClient  *client,
+                            const gchar         *uuid,
+                            GAsyncReadyCallback  callback,
+                            gpointer             user_data)
+{
+  EOSHackSoundClientPrivate *priv;
+  GTask *task;
+  g_return_if_fail (EOS_IS_HACK_SOUND_CLIENT (client));
+
+  priv = eos_hack_sound_client_get_instance_private (client);
+  task = g_task_new (client, NULL, callback, user_data);
+
+  g_dbus_proxy_call (priv->proxy,
+                     "StopSound", g_variant_new ("(s)", uuid),
+                     G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+                     (GAsyncReadyCallback) stop_sound_cb, task);
+}
+
+static void
+stop_sound_cb (GDBusProxy   *proxy,
+               GAsyncResult *res,
+               GTask        *task)
+{
+  GError *error = NULL;
+
+  g_dbus_proxy_call_finish (proxy, res, &error);
+  if (error)
+    {
+      g_task_return_error (task, error);
+      return;
+    }
+  g_task_return_pointer (task, NULL, NULL);
+}
+
+gpointer
+eos_hack_sound_client_stop_finish (EOSHackSoundClient  *client,
+                                   GAsyncResult        *result,
+                                   GError             **error)
+{
+  g_return_val_if_fail (g_task_is_valid (result, client), NULL);
+
+  return g_task_propagate_pointer (G_TASK (result), error);
+}
+
+void
+eos_hack_sound_client_play (EOSHackSoundClient  *client,
+                            const gchar         *sound_event_id,
+                            GAsyncReadyCallback  callback,
+                            gpointer             user_data)
+{
+  EOSHackSoundClientPrivate *priv;
+  GTask *task;
+  g_return_if_fail (EOS_IS_HACK_SOUND_CLIENT (client));
+
+  priv = eos_hack_sound_client_get_instance_private (client);
+  task = g_task_new (client, NULL, callback, user_data);
+
+  g_dbus_proxy_call (priv->proxy,
+                     "PlaySound", g_variant_new ("(s)", sound_event_id),
+                     G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+                     (GAsyncReadyCallback) play_sound_cb, task);
+}
+
+gchar *
+eos_hack_sound_client_play_finish (EOSHackSoundClient * client,
+    GAsyncResult * result, GError ** error)
+{
+  g_return_val_if_fail (g_task_is_valid (result, client), NULL);
+
+  return g_task_propagate_pointer (G_TASK (result), error);
+}
+
+static void
+play_sound_cb (GDBusProxy   *proxy,
+               GAsyncResult *res,
+               GTask        *task)
```

Comment:
This function leaks `task`. I suggest:

```c
static void
￼ play_sound_cb (GObject *source,
￼                GAsyncResult *res,
￼                gpointer data)
{
  GDBusProxy *proxy = G_DBUS_PROXY (source);
  g_autoptr(GTask) task = G_TASK (data);
```

which will unref it on the way out, and means you don't have to cast the callback in the call to `g_dbus_proxy_call`.